core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -639,6 +658,14 @@ public LocalExecutionPlan plan(
         return new LocalExecutionPlan(context.getDriverFactories(), partitionedSourceOrder);
     }
 
+    private int getTaskCount(PartitioningScheme partitioningScheme)
+    {
+        int[] bucketToPartition = partitioningScheme.getBucketToPartition()
```

Comment:
I think this wont handle skew if there is just hive bucketing (e.g 3) and there are more worker nodes (e.g. 10). Then it would identify `taskCount` as 3.

Maybe it's possible to fix somehow, e.g. bump bucket count in `io.trino.sql.planner.NodePartitioningManager#getNodePartitioningMap` in case scaling is on.

We can address it later on (TODO)

Maybe something like:
```
task_count = 4, bucket_count = 2, virtual_bucket_count = bucket * 2

     bucket, virtual_bucket, node
0 => 0,      0,              0
1 => 1,      1,              1
2 => 0,      2,              2
3 => 1,      3,              3
```